functions/controllers/kaiAIController.js

```diff
@@ -12,21 +12,20 @@ const app = express();
 
 const DEBUG = process.env.DEBUG;
 /**
- * Simulates communication with a Kai AI endpoint.
+ * Communicates with a Kai AI endpoint.
  *
- * @param {object} payload - The properties of the communication.
- * @param {object} props.data - The payload data object used in the communication.
- *  @param {Array} props.data.messages - An array of messages for the current user chat session.
- *  @param {object} props.data.user - The user object.
- *    @param {string} props.data.user.id - The id of the current user.
- *    @param {string} props.data.user.fullName - The user's full name.
- *    @param {string} props.data.user.email - The users email.
- *  @param {object} props.data.tool_data - The payload data object used in the communication.
- *    @param {string} props.data.tool_data.tool_id - The payload data object used in the communication.
- *    @param {Array} props.data.tool_data.inputs - The different form input values sent for a tool.
- *  @param {string} props.data.type - The payload data object used in the communication.
- *
- * @return {object} The response from the AI service.
+ * @async
+ * @function kaiCommunicator
+ * @param {Object} payload - The properties of the communication.
+ * @param {Object} payload.data - The payload data object used in the communication.
+ * @param {Array} [payload.data.messages] - An array of messages for the current user chat session.
+ * @param {Object} payload.data.user - The user object.
+ * @param {Object} [payload.data.tool_data] - The tool-specific data object.
+ * @param {string} payload.data.type - The type of communication (e.g., 'chat', 'tool').
+ * @returns {Promise<Object>} A promise that resolves to an object with status and data properties.
+ * @returns {string} return.status - The status of the operation ('success').
+ * @returns {Object} return.data - The data returned from the AI service.
+ * @throws {HttpsError} If there's an error in communication or processing.
```

Comment:
I would refrain from editing the docs that were already there as they may have had different meanings than what you are trying to mean with your edits. Let Billy go through it and confirm this.

Reply:
Undo modification